app/models/docker_builder_service.rb

```diff
@@ -32,6 +35,31 @@ def run!(image_name: nil, push: false, tag_as_latest: false)
     JobExecution.start_job(job_execution)
   end
 
+  def run_build_image_job(local_job, image_name, push: false, tag_as_latest: false)
```

Comment:
I would prefer to do this as well but as I said in the earlier comment, this piece of code has a coupling with ```DockerBuilderService``` in parsing the final output and to certain extent kinda of intertwine a local running job and a remote running job with logs content and job status passed around.